textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/queryrewriter/WordBase.java

```diff
@@ -0,0 +1,35 @@
+package edu.uci.ics.textdb.dataflow.queryrewriter;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.URL;
+import java.util.HashSet;
+
+/**
+ * Created by shiladityasen on 4/30/16.
+ * Class to provide simple abstraction of a dictionary of words with constant retrieval time
+ * Used by FuzzyTokenizer to process query
+ */
+public class WordBase
```

Comment:
No. The functionalities of both `Dictionary` and `WordBase` is to store the list of words. Moreover 'Dictionary' class needs a constructor which accepts a source file name. We didn't implement it now, but we may need it later. 
The only extra method which DictionaryMatcher doesn't need is `contains()` method.